pkg/tasks/update_template_content.go

```diff
@@ -329,3 +348,215 @@ func getRHRepoContentPath(rawURL string) (string, error) {
 	}
 	return u.Path[1 : len(u.Path)-1], nil
 }
+
+// RunCandlepin creates an environment for the template and content sets for each repository.
+// Each content set's URL is the distribution URL created during RunPulp().
+// May promote or demote content from the environment, depending on if the repository is being added or removed from template.
+// If not created for the given org previously, this will also create a product and pool.
+func (t *UpdateTemplateContent) RunCandlepin() error {
+	var err error
+
+	err = t.cpClient.CreateProduct(t.ownerKey)
+	if err != nil {
+		return err
+	}
+
+	poolID, err := t.cpClient.CreatePool(t.ownerKey)
+	if err != nil {
+		return err
+	}
+	t.payload.PoolID = &poolID
+	err = t.updatePayload()
+	if err != nil {
+		return err
+	}
+
+	content, contentIDs, err := t.getContentList()
+	if err != nil {
+		return err
+	}
+
+	for _, item := range content {
+		err = t.cpClient.CreateContent(t.ownerKey, item)
+		if err != nil {
+			return err
+		}
+	}
+
+	// TODO we can use create content batch when the api spec is fixed
+	//err = c.client.CreateContentBatch(candlepin_client.DevelOrgKey, content)
+	//if err != nil {
+	//	return err
+	//}
+
+	err = t.cpClient.AddContentBatchToProduct(t.ownerKey, contentIDs)
+	if err != nil {
+		return err
+	}
+
+	contentPath, err := t.pulpClient.GetContentPath()
+	if err != nil {
+		return err
+	}
+
+	envID := candlepin_client.GetEnvironmentID(t.payload.TemplateUUID)
+	env, err := t.createOrFetchEnvironment(envID, contentPath)
```

Comment:
I'm not sure how well I described the pathing in the jira issue, but because of the way pulp domains work, we actually want to set the prefix of the environment to the red hat domain's content path.  

We then plan to use environment content overrides (which was just merged in candlepin) to override the content paths for the custom repos.  So it will look a little something like this

Prefix:   https://pulp.content.server.com/pulp/RH_DOMAIN/content/
RH Content (which already exists in candlepin) path:  /content/dist/rhel8/8/x86_64/baseos/os

Which means we can't just simply expose custom content in the same way, and custom repos will be broken right now.  The plan is to use content overrides, to override the path to be:

Override URL:  https://pulp.content.server.com/pulp/ORG_DOMAIN/content/path/to/content

This environment override was *just* merged in candlepin in 4.4.6, and i figured we'd just leave custom content broken here, but we could go ahead and integrate that here too, or still do it as a followup PR.

Reply:
I didn't add content overrides, but I think the prefix is correct now